seqr/views/status_tests.py

```diff
@@ -2,61 +2,86 @@
 from django.urls.base import reverse
 import mock
 from requests import HTTPError
+import responses
 
 from seqr.views.status import status_view
 from seqr.utils.search.elasticsearch.es_utils_tests import urllib3_responses
 
 
-class StatusTest(TestCase):
+class StatusTest(object):
 
-    def _test_status_error(self, url, mock_logger, es_error):
+    def _test_status_error(self, url, mock_logger):
         response = self.client.get(url)
         self.assertEqual(response.status_code, 400)
         self.assertDictEqual(
             response.json(), {'version': 'v1.0', 'dependent_services_ok': False, 'secondary_services_ok': False})
-        mock_logger.error.assert_has_calls([
+        calls = [
             mock.call('Database "default" connection error: No connection'),
             mock.call('Database "reference_data" connection error: No connection'),
             mock.call('Redis connection error: Bad connection'),
-            mock.call(f'Search backend connection error: {es_error}'),
-            mock.call('Kibana connection error: Connection refused: HEAD /status'),
-        ])
+            mock.call(f'Search backend connection error: {self.SEARCH_BACKEND_ERROR}'),
+        ]
+        if self.HAS_KIBANA:
+            calls.append(mock.call('Search Admin connection error: Connection refused: HEAD /status'))
+        mock_logger.error.assert_has_calls(calls)
         mock_logger.reset_mock()
 
-    @mock.patch('seqr.utils.search.elasticsearch.es_utils.ELASTICSEARCH_SERVICE_HOSTNAME', 'testhost')
     @mock.patch('seqr.views.status.redis.StrictRedis')
     @mock.patch('seqr.views.status.connections')
     @mock.patch('seqr.views.status.logger')
     @urllib3_responses.activate
+    @responses.activate
     def test_status(self, mock_logger, mock_db_connections, mock_redis):
         url = reverse(status_view)
 
         mock_db_connections.__getitem__.return_value.cursor.side_effect = Exception('No connection')
         mock_redis.return_value.ping.side_effect = HTTPError('Bad connection')
+        responses.add(responses.GET, 'http://test-hail:5000/status', status=400)
 
-        self._test_status_error(url, mock_logger, es_error='No response from elasticsearch ping')
-
-        with mock.patch('seqr.utils.search.elasticsearch.es_utils.ELASTICSEARCH_SERVICE_HOSTNAME', ''):
-            self._test_status_error(url, mock_logger, es_error='Elasticsearch backend is disabled')
+        self._test_status_error(url, mock_logger)
 
         mock_db_connections.__getitem__.return_value.cursor.side_effect = None
         mock_redis.return_value.ping.side_effect = None
+        responses.add(responses.GET, 'http://test-hail:5000/status', status=200)
         urllib3_responses.add(urllib3_responses.HEAD, '/', status=200)
         urllib3_responses.add(urllib3_responses.HEAD, '/status', status=500)
```

Comment:
Can the responses setting be moved to the child classes? So that we don't have to mix the setting in the shared base class.